WordPress/gutenberg
# Font Library: font collection refactor to use the new schema

## What?
- Refactors of the font collection functionality to implement the new schema defined for the data.
- Make Font Collection Rest Controller more like other WordPress controllers
- Updates the google fonts collection data URL.
- Improves the unit tests

## Why?
To be more consistent with other WordPress controllers.

## How?
Refactoring frontend and backend to consume the new shape of the data.

## Testing Instructions
- Navigate the font library and check that you can browse the Google Fonts collection.
- Check that you can install fonts from the default Google Fonts collection.
